h2o-extensions/xgboost/src/main/java/ml/dmlc/xgboost4j/java/XGBoostSetupTask.java

```diff
@@ -38,6 +40,15 @@ protected void execute() {
     final DMatrix matrix;
     try {
       matrix = makeLocalMatrix();
+      if (_parms._save_matrix_directory != null) {
+        File directory = new File(_parms._save_matrix_directory);
+        if (directory.mkdirs()) {
```

Comment:
This might return false if the directory already exists, but it will also return false if the directory could not be created. Something like (quickly written):

```java
        if(directory.exists()){
          // continue
        }
        else if (directory.mkdirs()) {
          Log.debug("Created directory for matrix export: " + directory.getAbsolutePath());
        } else{
          throw new IllegalStateException("Unable to create directory for matrix export: " + directory.getAbsolutePath());
        }
```

Reply:
hmm, I think it is okay :)

Reply:
Fixed